addons/mail/static/src/messaging/component/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1014,6 +1017,70 @@ QUnit.test('open 3 different chat windows: not enough screen width', async funct
     );
 });
 
+QUnit.test('chat window: switch on TAB', async function (assert) {
+    assert.expect(7);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                name: "channel1",
+            }, {
+                channel_type: "channel",
+                id: 2,
+                name: "channel2",
+            }, {
+                channel_type: "channel",
+                id: 3,
+                name: "channel3",
+            }],
+        },
+    });
+    await this.start({
+        debug: true,
+    });
+    this.env.entities.Thread.find(thread => thread.id === 1).open();
+    await afterNextRender();
```

Comment:
Prefer wrapping statements that are expected to trigger a re-render inside function as parameter of `afterNextRender`.

Reason to prefer this syntax is that it is much more robust and ensures `afterNextRender()` correctly resolve after action. Here, it's possible that open triggers a re-render already and `afterNextRender()` is never resolved and crash as a result.

```js
await afterNextRender(() => this.env.entities.Thread.find(thread => thread.id === 1).open());
```

(could be multi-line)